konfork-core/src/commonMain/kotlin/io/github/konfork/core/validators/Strings.kt

```diff
@@ -6,70 +6,101 @@ import io.github.konfork.core.Specification
 import io.github.konfork.core.stringHint
 import io.github.konfork.predicates.*
 
-fun <C, E> Specification<C, String, E>.minLength(hint: HintBuilder<C, String, E>, length: Int): ConstraintBuilder<C, String, E> {
-    require(length >= 0) { IllegalArgumentException("minLength requires the length to be >= 0") }
-    return addConstraint(hint, length) { it.length >= length }
-}
-fun <C> Specification<C, String, String>.minLength(length: Int) =
-    minLength(stringHint("must have at least {0} characters"), length)
+fun <C, E> Specification<C, String, E>.all(hint: HintBuilder<C, String, E>, predicate: (Char) -> Boolean) =
+    addConstraint(hint) { it.all(predicate) }
 
-fun <C, E> Specification<C, String, E>.maxLength(hint: HintBuilder<C, String, E>, length: Int): ConstraintBuilder<C, String, E> {
-    require(length >= 0) { IllegalArgumentException("maxLength requires the length to be >= 0") }
-    return addConstraint(hint, length) { it.length <= length }
-}
+fun <C> Specification<C, String, String>.all(predicate: (Char) -> Boolean) =
+    all(stringHint("not all characters comply"), predicate)
 
-fun <C> Specification<C, String, String>.maxLength(length: Int) =
-    maxLength(stringHint("must have at most {0} characters"), length)
+fun <C, E> Specification<C, String, E>.any(hint: HintBuilder<C, String, E>, predicate: (Char) -> Boolean) =
+    addConstraint(hint) { it.any(predicate) }
 
-fun <C, E> Specification<C, String, E>.lengthIn(hint: HintBuilder<C, String, E>, range: IntRange): ConstraintBuilder<C, String, E> {
-    require(range.first >= 0 && range.last >= 0) { IllegalArgumentException("lengthIn requires the length to be >= 0") }
-    return addConstraint(hint, range.first, range.last) { it.length in range }
-}
+fun <C> Specification<C, String, String>.any(predicate: (Char) -> Boolean) =
+    any(stringHint("none of the characters comply"), predicate)
```

Comment:
In many places it might be better to use `staticHint` instead of `stringHint`

Reply:
:+1: